crypto/fipsmodule/rand/urandom.c

```diff
@@ -289,7 +357,13 @@ static void wait_for_entropy(void) {
       break;
     }
 
-    usleep(250000);
+    int r;
+    struct timespec sleep_time = {.tv_sec = 0, .tv_nsec = MILLISECONDS_250 };
+
+    do {
```

Comment:
Do we need the loop here or just call nanosleep() once in the same way as usleep was called?

Reply:
`nanosleep` can be interrupted and other system faults. We do need to retry it. You should really also retry `usleep` imo.

Reply:
Do we want to loop indefinitely without a cap?

Reply:
Implemented best-effort to avoid blocking.